### PR TITLE
Bugfix for weekday validator

### PR DESCRIPTION
**Pull request by philodavies**
Output from go test:
```
--- FAIL: TestNextTickAfter (0.01s)
    --- FAIL: TestNextTickAfter/next_run_after (0.01s)
        --- FAIL: TestNextTickAfter/next_run_after/next_run_after_excl_#63:_0_0_*_*_1#1 (0.00s)
            next_test.go:66: [0 0 * * 1#1][2009-11-23 00:00:00] expected 2009-12-07 00:00:00, got 2010-01-04 00:00:00
```

I haven't had time to dig into why this case is not working as expected, but I noticed something really weird here. If I say the first Monday of the month it doesn't always give the right answer:

Go Playground example: https://go.dev/play/p/0GJnCNPWCWC

### CODE REVIEW

**Comment by philodavies** [comment]
Actually got a better repro.. Anytime the first x-day of the month is the 7th, it gets skipped
https://go.dev/play/p/yoI2lcTtyRY

File: validator.go [hunk 1]
```diff
@@ -142,5 +142,5 @@ func isValidWeekDay(val string, last int, ref time.Time) (bool, error) {
 		return false, nil
 	}
 
-	return ref.Day()/7 == nth-1, nil
+	return (ref.Day()-1)/7 == nth-1, nil
```
Comment (adhocore) [inline validator.go:145]: strange this change still has all tests passed

Comment (adhocore) [inline validator.go:145]: strange this change still has all tests passed

Comment (philodavies) [reply validator.go:145]: I think this is an edge case that wasn't covered in the previous unit tests. If we use a cron string that does first/second/etc monday/tuesday/etc of the month, then we get in a situation where days 1-6 are in week 0, but day 7 is week 1. This change makes it that days 1-7 are in week 0.

**Review by adhocore — APPROVED** [review]
